lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,576 @@
+                        ev = 'S0'
+                    elif ev == 's1':
+                        ev = 'S1'
+                    elif ev == 's2':
+                        ev = 'S2'
+                    elif ev == 's3':
+                        ev = 'S3'
+                    elif ev == 'p1':
+                        ev = 'P1'
+                    elif ev == 'p2':
+                        ev = 'P2'
+                    elif ev == 'p3':
+                        ev = 'P3'
+                    elif ev == 'p4':
+                        ev = 'P4'
+                    elif ev == 'p6':
+                        ev = 'P6'
+                    elif ev == 'p11':
+                        ev = 'P11'
+                    elif ev == 'p15':
+                        ev = 'P15'
+                    elif ev == 'system':
+                        ev = 'System'
+                    elif ev == 'system2':
+                        ev = 'System2'
+                    elif ev == 'elastic_pool':
+                        ev = 'ElasticPool'
+                    self.parameters["requested_service_objective_name"] = ev
+                elif key == "elastic_pool_name":
+                    self.parameters["elastic_pool_name"] = kwargs[key]
+                elif key == "read_scale":
+                    ev = kwargs[key]
+                    if ev == 'enabled':
+                        ev = 'Enabled'
+                    elif ev == 'disabled':
+                        ev = 'Disabled'
+                    self.parameters["read_scale"] = ev
+                elif key == "sample_name":
+                    ev = kwargs[key]
+                    if ev == 'adventure_works_lt':
+                        ev = 'AdventureWorksLT'
+                    self.parameters["sample_name"] = ev
+                elif key == "zone_redundant":
+                    self.parameters["zone_redundant"] = kwargs[key]
+
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if "location" not in self.parameters:
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_sqldatabase()
+
+        if not old_response:
+            self.log("SQL Database instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("SQL Database instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                self.log("Need to check if SQL Database instance has to be deleted or may be updated")
+                self.to_do = Actions.Update
```

Comment:
This is the same as my comment in the postgresql review, we need better checking here to determine if an update is required. Right now if the database exists and state is present and we are running in check mode, a change will be reported which is not correct.